src/components/Pagination.jsx

```diff
@@ -29,7 +30,7 @@ export const Pagination = ({
   }, [itemsPerPage, listOfItems]);
 
   useEffect(() => {
-    const itemsListOnCurrentPage = getListItemsOnCurrentPage(activePageIDState);
+    const itemsListOnCurrentPage = getListItemsOnCurrentPage(currentPageState);
     setCurrentPageItems(itemsListOnCurrentPage);
 
     function getListItemsOnCurrentPage(activePageIDState) {
```

Comment:
You could move this function out of this component and make it a pure util (that doesn't rely on data outside of its scope, and only takes params and returns a value).